nix/default.nix

```diff
@@ -1,53 +1,54 @@
+{ system ? builtins.currentSystem }:
 let
-  pkgs = import ./nixpkgs.nix {
+  pkgs = (import ./nixpkgs.nix {
     config = {
       packageOverrides = pkg: {
         go_1_12 = pkg.go_1_14;
+        gpgme = (static pkg.gpgme);
+        libassuan = (static pkg.libassuan);
+        libgpgerror = (static pkg.libgpgerror);
+        libseccomp = (static pkg.libseccomp);
+        systemd = pkg.systemd.overrideAttrs(x: {
+          mesonFlags = x.mesonFlags ++ [ "-Dstatic-libsystemd=true" ];
+          postFixup = ''
+            ${x.postFixup}
+            sed -ri "s;$out/(.*);$nukedRef/\1;g" $lib/lib/libsystemd.a
+          '';
+        });
       };
     };
-  };
+  });
 
-  static = pkg: pkg.overrideAttrs(old: {
-    configureFlags = (old.configureFlags or []) ++
+  static = pkg: pkg.overrideAttrs(x: {
+    configureFlags = (x.configureFlags or []) ++
       [ "--without-shared" "--disable-shared" ];
     dontDisableStatic = true;
     enableSharedExecutables = false;
     enableStatic = true;
   });
 
-  patchLvm2 = pkg: pkg.overrideAttrs(old: {
-    configureFlags = [
-      "--disable-cmdlib" "--disable-readline" "--disable-udev_rules"
-      "--disable-udev_sync" "--enable-pkgconfig" "--enable-static_link"
-    ];
-    preConfigure = old.preConfigure + ''
-      substituteInPlace libdm/Makefile.in --replace \
-        SUBDIRS=dm-tools SUBDIRS=
-      substituteInPlace tools/Makefile.in --replace \
-        "TARGETS += lvm.static" ""
-      substituteInPlace tools/Makefile.in --replace \
-        "INSTALL_LVM_TARGETS += install_tools_static" ""
-    '';
-    postInstall = "";
-  });
-
-  self = {
-    podman-static = (pkgs.podman.overrideAttrs(old: {
+  self = with pkgs; {
+    podman-static = (podman.overrideAttrs(x: {
       name = "podman-static";
-      buildInputs = old.buildInputs ++ (with pkgs; [
-        (static pkgs.libassuan)
-        (static pkgs.libgpgerror)
+      src = ./..;
+      doCheck = false;
+      buildInputs = [
         git
         glibc
         glibc.static
-      ]);
-      src = ./..;
-      EXTRA_LDFLAGS = ''-linkmode external -extldflags "-static -lm"'';
-      BUILDTAGS = ''static netgo apparmor selinux seccomp systemd varlink containers_image_ostree_stub'';
-    })).override {
-      gpgme = (static pkgs.gpgme);
-      libseccomp = (static pkgs.libseccomp);
-      lvm2 = (patchLvm2 (static pkgs.lvm2));
-    };
+        gpgme
+        libapparmor
+        libassuan
+        libgpgerror
+        libseccomp
+        libselinux
+        systemd
+      ];
+      prePatch = ''
+        export LDFLAGS='-static-libgcc -static'
+        export EXTRA_LDFLAGS='-linkmode external -extldflags "-static -lm"'
+        export BUILDTAGS='static netgo varlink apparmor selinux seccomp systemd btrfs_noversion exclude_graphdriver_btrfs exclude_graphdriver_devicemapper'
```

Comment:
We removed the devicemapper support. Was this intentional and is it okay for podman?

Reply:
We've been compiling it out by default for a long time now - a few years ago, it was unsafe to use in a multiprocess environment. I've heard that the situation has improved since then, but we never reenabled it.

Reply:
As of today my Ansible Role for Podman also NOT compile with btrfs or LVM2 support (https://github.com/alvistack/ansible-role-podman/blob/2aebee3/tasks/main.yml#L83-L107).

This had been tested with Vagrant + KVM on Ubuntu 18.04+ / CentOS 7+ / openSUSE 15.1 / Debian 10 / Fedora 32, which all looks fine with default installation (https://travis-ci.com/github/alvistack/ansible-role-podman/builds/168093040).

Reply:
Sounds good to me